### PR TITLE
Implement suspend/resume functionality

### LINKED ISSUE #3: [UTM] Auto-pausing on task switch
Implement hooks to pause VM on task switch and low-memory conditions.

### LINKED ISSUE #24: [UTM] Handle memory warning
App gets killed if too much memory is used. We should handle it better.

### LINKED ISSUE #98: [UI] Disable backing out of VM
Alternative to #6 where we just assume nobody needs to start two vms and only terminate on completion. This is probably a lot easier to do and the immense amount of work it takes to implement cleanup is not worth it.

### LINKED ISSUE #107: Pause/resume VM

### PR DESCRIPTION
**Pull request by osy86**
- [x] Backend for create/restoring snapshots
- [x] UTM manager: handle suspend/resume (resolves #107)
- [x] UI for pause/resume button
- [x] Refactor UI code to disable backing out of VM view (resolves #98)
- [x] Auto pause on task switch (resolves #3) 
- [x] Prompt pause on memory pressure (resolves #24)

### CODE REVIEW

File: Configuration/UTMViewState.m [hunk 1]
```diff
@@ -78,6 +79,14 @@ - (void)setShowKeyboard:(BOOL)showKeyboard {
     _rootDict[kUTMViewStateShowKeyboardKey] = [NSNumber numberWithBool:showKeyboard];
 }
 
+- (BOOL)suspended {
+    return [_rootDict[kUTMViewStateSuspendedKey] boolValue];
+}
+
+- (void)setSuspended:(BOOL)suspended {
+    _rootDict[kUTMViewStateSuspendedKey] = [NSNumber numberWithBool:suspended];
```
Comment (saagarjha) [inline Configuration/UTMViewState.m]: `@(suspended)`?

Comment (osy86) [reply Configuration/UTMViewState.m]: I didn't know you can do that.

Comment (osy86) [reply Configuration/UTMViewState.m]: Is there a shorthand for the reverse: `[_rootDict[kUTMViewStateSuspendedKey] boolValue]`?

Comment (saagarjha) [reply Configuration/UTMViewState.m]: Not that I know of, sadly :(

**Pull request merged**